.test-defs/cmd/create-shoot/main.go

```diff
@@ -141,11 +141,61 @@ func main() {
 		testLogger.Fatalf("Cannot create ShootGardenerTest %s", err.Error())
 	}
 
+	gardenerTestOperation, err := framework.NewGardenTestOperation(ctx, shootGardenerTest.GardenClient, testLogger, nil)
+	if err != nil {
+		testLogger.Fatalf("Cannot create gardener test operation %s: %s", shootName, err.Error())
+	}
+
 	_, shootObject, err := framework.CreateShootTestArtifacts(shootArtifactPath, "", true)
 	if err != nil {
 		testLogger.Fatalf("Cannot create shoot artifact %s", err.Error())
 	}
 
+	modifyShootObject(shootObject)
+
+	testLogger.Infof("Create shoot %s in namespace %s", shootName, projectNamespace)
+	if err := printShoot(shootObject); err != nil {
+		testLogger.Fatalf("Cannot decode shoot %s: %s", shootName, err)
+	}
+
+	shootGardenerTest.Shoot = shootObject
+	shootObject, err = shootGardenerTest.CreateShoot(ctx)
+	if err != nil {
+		if shootObject != nil {
+			if err := gardenerTestOperation.AddShoot(ctx, shootObject); err != nil {
+				testLogger.Errorf("Cannot add shoot %s to test operation: %s", shootName, err.Error())
+			}
+			gardenerTestOperation.DumpState(ctx)
+		}
+		testLogger.Fatalf("Cannot create shoot %s: %s", shootName, err.Error())
+	}
+	testLogger.Infof("Successfully created shoot %s", shootName)
+
+	if err := gardenerTestOperation.AddShoot(ctx, shootObject); err != nil {
```

Comment:
Is it needed as we already add it at line 165?

Reply:
yes because line 165 handles the error case of `shootGardenerTest.CreateShoot(ctx)`. So there we want to print the state if the create operation was not successful but the shoot is still accessible (reconciliation errors, cloudprovider issues, etc.).